From 1a0fca6511cceb8e19069e310bf29c1a33ee7fbf Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Sun, 9 Mar 2014 14:22:07 +0100
Subject: [PATCH 01/14] fix tputs()

---
 pdcurses/terminfo.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/pdcurses/terminfo.c b/pdcurses/terminfo.c
index c099ada7..2c503a84 100644
--- a/pdcurses/terminfo.c
+++ b/pdcurses/terminfo.c
@@ -211,5 +211,8 @@ int tputs(const char *str, int affcnt, int (*putfunc)(int))
 {
     PDC_LOG(("tputs() - called\n"));
 
-    return ERR;
+    while (*str)
+        putfunc(*str++);
+
+    return 0;
 }
-- 
2.39.1.windows.1

