From c0b2e904cf32bf30afdd76e02b8640c0a505a4ba Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Sun, 9 Mar 2014 14:45:02 +0100
Subject: [PATCH 04/14] add debug information in release build

---
 win32/gccwin32.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/win32/gccwin32.mak b/win32/gccwin32.mak
index 7504bb34..77a017b0 100644
--- a/win32/gccwin32.mak
+++ b/win32/gccwin32.mak
@@ -24,7 +24,7 @@ ifeq ($(DEBUG),Y)
 	CFLAGS  = -g -Wall -DPDCDEBUG
 	LDFLAGS = -g
 else
-	CFLAGS  = -O2 -Wall
+	CFLAGS  = -O2 -Wall -g
 	LDFLAGS =
 endif
 
-- 
2.39.1.windows.1

