From 24fc0f743617fe35ae6201477b0e491d750885fc Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Sun, 4 Jan 2015 17:23:26 +0100
Subject: [PATCH 08/14] line up

---
 pdcurses/terminfo.c | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/pdcurses/terminfo.c b/pdcurses/terminfo.c
index e9959788..53c3fffd 100644
--- a/pdcurses/terminfo.c
+++ b/pdcurses/terminfo.c
@@ -197,6 +197,13 @@ char *tgetstr(const char *id, char **area)
         *area += 2;
         return val;
     }
+    else if (id[0] == 'u' && id[1] == 'p')
+    {
+        char *val = *area;
+        strcpy(val, "\013");
+        *area += 2;
+        return val;
+    }
 #endif
 
     return (char *)NULL;
@@ -284,6 +291,21 @@ void PDC_clear_page(void)
     SetConsoleWindowInfo(hout, TRUE, &sr);
     SetConsoleCursorPosition(hout, coord);
 }
+
+void PDC_line_up(void)
+{
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+    HANDLE hout = GetStdHandle(STD_OUTPUT_HANDLE);
+    GetConsoleScreenBufferInfo(hout, &csbi);
+    if (csbi.dwCursorPosition.Y <= 0)
+        return;
+
+    COORD coord = {
+        csbi.dwCursorPosition.X,
+        csbi.dwCursorPosition.Y - 1
+    };
+    SetConsoleCursorPosition(hout, coord);
+}
 #endif
 
 int tputs(const char *str, int affcnt, int (*putfunc)(int))
@@ -300,6 +322,11 @@ int tputs(const char *str, int affcnt, int (*putfunc)(int))
             PDC_clear_page();
             continue;
         }
+	else if (c == '\013')
+        {
+            PDC_line_up();
+            continue;
+        }
 #endif
         putfunc(c);
     }
-- 
2.39.1.windows.1

