From c368a1e6b8eae184495c97c1dea6badfe3663efd Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Wed, 2 Sep 2015 20:42:02 +0100
Subject: [PATCH 09/14] fix doupdate

---
 pdcurses/refresh.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pdcurses/refresh.c b/pdcurses/refresh.c
index 0b8e1ca5..21a37ef0 100644
--- a/pdcurses/refresh.c
+++ b/pdcurses/refresh.c
@@ -136,7 +136,6 @@ int doupdate(void)
 
     if (isendwin())         /* coming back after endwin() called */
     {
-        reset_prog_mode();
         clearall = TRUE;
         SP->alive = TRUE;   /* so isendwin() result is correct */
     }
-- 
2.39.1.windows.1

