From 356f798fe623096cf69d3c6533556d631651fecc Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Tue, 4 Mar 2014 20:11:21 +0100
Subject: [PATCH 04/23] Clear symbols if executable can't be attached

---
 gdb/exec.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gdb/exec.c b/gdb/exec.c
index c8c32ecc27..14b5a06aea 100644
--- a/gdb/exec.c
+++ b/gdb/exec.c
@@ -223,6 +223,12 @@ exec_file_locate_attach (int pid, int defer_bp_reset, int from_tty)
   do_cleanups (old_chain);
 }
 
+static void
+symbol_file_clear_cleanup (void *ignore)
+{
+  symbol_file_clear (0);
+}
+
 /* Set FILENAME as the new exec file.
 
    This function is intended to be behave essentially the same
@@ -267,6 +273,10 @@ exec_file_attach (const char *filename, int from_tty)
       int scratch_chan;
       struct target_section *sections = NULL, *sections_end = NULL;
       char **matching;
+      struct cleanup *symbol_cleanup;
+
+      /* Clear symbols if executable can't be attached.  */
+      symbol_cleanup = make_cleanup (symbol_file_clear_cleanup, 0);
 
       if (is_target_filename (filename))
 	{
@@ -378,6 +388,8 @@ exec_file_attach (const char *filename, int from_tty)
       /* Tell display code (if any) about the changed file name.  */
       if (deprecated_exec_file_display_hook)
 	(*deprecated_exec_file_display_hook) (filename);
+
+      discard_cleanups (symbol_cleanup);
     }
 
   bfd_cache_close_all ();
-- 
2.15.1.windows.2

