From d839242ed06ac13d36474420d215d8c0c4ede7fd Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Tue, 4 Mar 2014 20:15:24 +0100
Subject: [PATCH 05/23] Use thiscall calling convention for class members

---
 gdb/eval.c      |  2 ++
 gdb/i386-tdep.c | 30 +++++++++++++++++++++++++++++-
 2 files changed, 31 insertions(+), 1 deletion(-)

diff --git a/gdb/eval.c b/gdb/eval.c
index 6f74c41b9f..05d556f4e7 100644
--- a/gdb/eval.c
+++ b/gdb/eval.c
@@ -1177,6 +1177,8 @@ evaluate_funcall (type *expect_type, expression *exp, int *pos,
 	  nargs--;
 	  argvec++;
 	}
+      if (argvec[0])
+	TYPE_MAIN_TYPE (value_type (argvec[0]))->flag_static = static_memfuncp != 0;
     }
   else if (op == STRUCTOP_MEMBER || op == STRUCTOP_MPTR)
     {
diff --git a/gdb/i386-tdep.c b/gdb/i386-tdep.c
index 9ce329eb20..f33e5aa238 100644
--- a/gdb/i386-tdep.c
+++ b/gdb/i386-tdep.c
@@ -47,6 +47,8 @@
 #include "i386-tdep.h"
 #include "i387-tdep.h"
 #include "x86-xstate.h"
+#include "infcall.h"
+#include "producer.h"
 
 #include "record.h"
 #include "record-full.h"
@@ -2678,6 +2680,27 @@ i386_push_dummy_call (struct gdbarch *gdbarch, struct value *function,
   int i;
   int write_pass;
   int args_space = 0;
+  struct type *func_type = value_type (function);
+  int i386_windows_thiscall = 0;
+
+  if (func_type)
+    {
+      func_type = check_typedef (func_type);
+
+      if (TYPE_CODE (func_type) == TYPE_CODE_PTR)
+	func_type = check_typedef (TYPE_TARGET_TYPE (func_type));
+
+      if ((TYPE_CODE (func_type) == TYPE_CODE_METHOD)
+	  && nargs > 0
+	  && !TYPE_MAIN_TYPE (value_type (function))->flag_static)
+	{
+	  struct compunit_symtab *symtab =
+	    find_pc_compunit_symtab (find_function_addr (function, NULL));
+	  if (symtab
+	      && producer_is_gcc_ge_4 (symtab->producer) > 6)
+	    i386_windows_thiscall = 1;
+	}
+    }
 
   /* BND registers can be in arbitrary values at the moment of the
      inferior call.  This can cause boundary violations that are not
@@ -2707,7 +2730,7 @@ i386_push_dummy_call (struct gdbarch *gdbarch, struct value *function,
 	    args_space += 4;
 	}
 
-      for (i = 0; i < nargs; i++)
+      for (i = i386_windows_thiscall; i < nargs; i++)
 	{
 	  int len = TYPE_LENGTH (value_enclosing_type (args[i]));
 
@@ -2759,6 +2782,11 @@ i386_push_dummy_call (struct gdbarch *gdbarch, struct value *function,
   /* ...and fake a frame pointer.  */
   regcache_cooked_write (regcache, I386_EBP_REGNUM, buf);
 
+  /* 'this' pointer needs to be in ECX.  */
+  if (i386_windows_thiscall)
+    regcache_cooked_write (regcache, I386_ECX_REGNUM,
+			   value_contents_all (args[0]));
+
   /* MarkK wrote: This "+ 8" is all over the place:
      (i386_frame_this_id, i386_sigtramp_frame_this_id,
      i386_dummy_id).  It's there, since all frame unwinders for
-- 
2.15.1.windows.2

