From d400b9607ffad5175614b72e8ccf27ee7d3ede32 Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Tue, 11 Mar 2014 19:50:44 +0100
Subject: [PATCH 10/23] Fix resize crashes

---
 gdb/tui/tui-win.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/gdb/tui/tui-win.c b/gdb/tui/tui-win.c
index f3b7873d53..9fb663d68e 100644
--- a/gdb/tui/tui-win.c
+++ b/gdb/tui/tui-win.c
@@ -778,12 +778,7 @@ tui_resize_all (void)
       split_diff = height_diff / num_wins_displayed;
       cmd_split_diff = split_diff;
       if (height_diff % num_wins_displayed)
-	{
-	  if (height_diff < 0)
-	    cmd_split_diff--;
-	  else
-           cmd_split_diff++;
-       }
+	cmd_split_diff += height_diff % num_wins_displayed;
       /* Now adjust each window.  */
       /* erase + clearok are used instead of a straightforward clear as
          AIX 5.3 does not define clear.  */
@@ -874,6 +869,7 @@ tui_resize_all (void)
 
 	  /* Change the command window's height/width.  */
 	  TUI_CMD_WIN->generic.origin.y = locator->origin.y + 1;
+	  TUI_CMD_WIN->generic.width += width_diff;
 	  make_invisible_and_set_new_height (TUI_CMD_WIN,
 					     TUI_CMD_WIN->generic.height
 					     + cmd_split_diff);
-- 
2.15.1.windows.2

