From 87ee1819d3278ed741ef4e991823ccbe4e339f6e Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Thu, 1 Jan 2015 15:18:08 +0100
Subject: [PATCH 12/23] Use ctrl-left/right to move to previous/next word

---
 readline/readline.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/readline/readline.c b/readline/readline.c
index 071e1aa082..d01cee7f43 100644
--- a/readline/readline.c
+++ b/readline/readline.c
@@ -1163,6 +1163,9 @@ bind_arrow_keys_internal (map)
   rl_bind_keyseq_if_unbound ("\340O", rl_end_of_line);
   rl_bind_keyseq_if_unbound ("\340S", rl_delete);
   rl_bind_keyseq_if_unbound ("\340R", rl_overwrite_mode);
+
+  rl_bind_keyseq_if_unbound ("\340s", rl_backward_word);  /* ctrl-left  */
+  rl_bind_keyseq_if_unbound ("\340t", rl_forward_word);   /* ctrl-right */
 #endif
 
   _rl_keymap = xkeymap;
-- 
2.15.1.windows.2

