From e115c5a0c0d4bc57a602a0c3d36bd824434f1b51 Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Sun, 26 Jun 2016 14:57:33 +0200
Subject: [PATCH 21/23] Fix readline assert

---
 gdb/event-loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/event-loop.c b/gdb/event-loop.c
index 7934ab4c82..00c25614e7 100644
--- a/gdb/event-loop.c
+++ b/gdb/event-loop.c
@@ -319,7 +319,7 @@ gdb_do_one_event (void)
 	case 1:
 	  /* Are there events already waiting to be collected on the
 	     monitored file descriptors?  */
-	  res = gdb_wait_for_event (0);
+	  res = 0;
 	  break;
 	case 2:
 	  /* Are there any asynchronous event handlers ready?  */
-- 
2.15.1.windows.2

