From 1340538887fd44981882b2c8b25c5a59056f4b49 Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Fri, 16 Feb 2018 14:00:10 +0100
Subject: [PATCH 24/24] No segment registers for win64

---
 gdb/arch/amd64.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gdb/arch/amd64.c b/gdb/arch/amd64.c
index bd3df8b91d..e9636a8063 100644
--- a/gdb/arch/amd64.c
+++ b/gdb/arch/amd64.c
@@ -55,8 +55,10 @@ amd64_create_target_description (uint64_t xcr0, bool is_x32, bool is_linux)
 
   regnum = create_feature_i386_64bit_sse (tdesc, regnum);
   if (is_linux)
-    regnum = create_feature_i386_64bit_linux (tdesc, regnum);
-  regnum = create_feature_i386_64bit_segments (tdesc, regnum);
+    {
+      regnum = create_feature_i386_64bit_linux (tdesc, regnum);
+      regnum = create_feature_i386_64bit_segments (tdesc, regnum);
+    }
 
   if (xcr0 & X86_XSTATE_AVX)
     regnum = create_feature_i386_64bit_avx (tdesc, regnum);
-- 
2.15.1.windows.2

