--- boost/container_hash/hash.hpp	2018-12-05 20:58:16.000000000 +0100
+++ boost/container_hash/hash.hpp	2025-09-23 19:24:49.378000000 +0200
@@ -126,8 +126,13 @@ namespace boost
             typedef std::size_t result_type;
         };
 #else
+#ifndef __clang__
         template <typename T>
         struct hash_base : std::unary_function<T, std::size_t> {};
+#else
+        template <typename T>
+        struct hash_base : std::__unary_function<T, std::size_t> {};
+#endif
 #endif
 
         struct enable_hash_value { typedef std::size_t type; };
